app/controllers/user.js

```diff
@@ -44,6 +46,61 @@ exports.create = (req, res, next) => {
   
 };
 
+exports.signin = (req, res, next) => {
+
+  try {
+    validationResult(req).throw();
+  } catch (err) {
+    let errors = {};
+    if(err.mapped().email){
+      errors.email = err.mapped().email.msg + ';';
+    }
+    if(err.mapped().password){
+      errors.password = err.mapped().password.msg;
+    }
+    return res.status(400).send(errors);
+     
+  }
+
+  let input = emptyToNull(req.body);
+
+  if (req.headers.token){
+
+    let token = jwt.decode(req.headers.token, secret);
+
+    if(token.token == input.email){
+      return res.status(200).send('You are already logged in!');
+    }
+    
+  }
+
+  User.findOne({
+    where: {
+      email: input.email
+    }
+  }).then(result => {
+    
+    if(!result){
+      logger.info(`Login attempt with invalid email: "${input.email}"`);
+      return res.status(401).send('Database error');
```

Comment:
Same as above I would use some `invalidCredentialError` defined in the `error.js` file and use it in both cases, when we couldn't find the user in the db and when the user sent an invalid password. So you don't leak extra information to a potential intruder.